modules/querier/http.go

```diff
@@ -193,6 +193,7 @@ func (q *Querier) SearchHandler(w http.ResponseWriter, r *http.Request) {
 		http.Error(w, err.Error(), http.StatusInternalServerError)
 		return
 	}
+	w.Header().Set(api.HeaderContentType, api.HeaderAcceptJSON)
```

Comment:
set content type in `TraceByIDHandler` as well?

Reply:
Mhm, that's a good point 🤔  The TraceByID path is a bit odd because eventually it all reaches the `newTraceByIdMiddleware` in `frontend.go` where the payload is serialized according to the `Accept` header. I had to modify the `tracebyidsharding.go` because the tests weren't passing, but as it is, the tests are all passing. So while it would make sense to also add it to the `TraceByIDHandler` I also don't know if it'll make any practical difference.

Reply:
> where the payload is serialized according to the Accept header
Sadly I think `tracebyidsharding.go` makes no checks and [always assumes](https://github.com/grafana/tempo/blob/7d26ee26986411cf096f4a93ad1dd7fa422fa972/modules/frontend/tracebyidsharding.go#L120-L121) that the returned payload is proto. I don't expect a fix for that in this PR.

It then uses the `Accept` header in `frontend.go` to determine how to marshal the response for the caller.

Since we are adding the `Content-Type` header to other responses here I would like to just slap it in the `TraceByIDHandler` as well, but don't feel obligated to adjust the frontend.

Reply:
Done 👍